src/Compilers/CSharp/Test/Emit2/Semantics/CollectionLiteralTests.cs

```diff
@@ -5644,6 +5667,2548 @@ private static void VerifyTypes(SemanticModel model, ExpressionSyntax expr, stri
+                    IEnumerator IEnumerable.GetEnumerator() => _array.GetEnumerator();
+                }
+                public struct MyCollectionBuilder
+                {
+                    public static MyCollection Create(ReadOnlySpan<object> items)
+                        => new MyCollection(items.ToArray());
+                }
+                """;
+            var comp = CreateCompilation(new[] { sourceA, s_constructibleCollectionAttributeDefinition }, targetFramework: TargetFramework.Net70);
+            var refA = AsReference(comp, useCompilationReference);
+
+            string sourceB = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        MyCollection x = [];
+                        x.Report();
+                        MyCollection y = [1, 2, 3];
+                        y.Report();
+                    }
+                }
+                """;
+            CompileAndVerify(new[] { sourceB, s_collectionExtensions }, references: new[] { refA }, targetFramework: TargetFramework.Net70, expectedOutput: "[], [1, 2, 3], ");
+        }
+
+        [CombinatorialData]
+        [ConditionalTheory(typeof(CoreClrOnly))]
+        public void CollectionBuilder_ObjectElementType_02(bool useCompilationReference)
+        {
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
+                public struct MyCollection<T> : IEnumerable
+                {
+                    public MyCollection(T[] array) { }
+                    public IEnumerator GetEnumerator() => default;
+                }
+                public class MyCollectionBuilder
+                {
+                    public static MyCollection<T> Create<T>(ReadOnlySpan<T> items) => default;
+                }
+                """;
+            var comp = CreateCompilation(new[] { sourceA, s_constructibleCollectionAttributeDefinition }, targetFramework: TargetFramework.Net70);
+            var refA = AsReference(comp, useCompilationReference);
+
+            string sourceB = """
+                #pragma warning disable 219
+                class Program
+                {
+                    static void Main()
+                    {
+                        MyCollection<object> x = [];
+                        MyCollection<int> y = [1, 2, 3];
+                        MyCollection<string> z = new();
+                    }
+                }
+                """;
+            comp = CreateCompilation(sourceB, references: new[] { refA }, targetFramework: TargetFramework.Net70);
+            comp.VerifyEmitDiagnostics(
+                // (6,34): error CS9181: Could not find an accessible 'Create' method with the expected signature.
+                //         MyCollection<string> x = [];
```

Comment:
slightly wonky as this line in the comment doesn't exist in the test code.

Reply:
also unclear why this doesn't work.

Reply:
The _iteration type_ of the collection is `object` rather than `T`, so the span should be `ReadOnlySpan<object>`. The updated error message hopefully makes that clearer now.